pkg/k8s/rule_translate_test.go

```diff
@@ -240,11 +240,17 @@ func (s *K8sSuite) TestGenerateToCIDRFromEndpoint(c *C) {
 		epIP2 + "/32",
 	})
 
+	// first round of deletes, make sure nothing is deleted
+	err = deleteToCidrFromEndpoint(rule, endpointInfo, false)
+	c.Assert(err, IsNil)
+	c.Assert(len(rule.ToCIDRSet), Equals, 2)
+
+	// second round of deletes, make sure everything is deleted
```

Comment:
Why does running `generateToCidrFromEndpoint()` multiple times take multiple references to the underlying CIDRs? Is that expected? How often will that function be run?

Reply:
This is actually a bit trickier to reason about, because populateEgress() -> generateToCidrFromEndpoint() is only ever run from `TranslateEgress`, which guarantees to run `depopulateEgress()` first before `populateEgress()`, so I'm not even sure if it's possible for the actual codepaths to trigger `generateToCidrFromEndpoint()` twice in a row like this unit test does.